src/python-common/ceph/deployment/drive_selection/selector.py

```diff
@@ -111,6 +111,12 @@ def assign_devices(self, device_filter):
         for disk in self.disks:
             logger.debug("Processing disk {}".format(disk.path))
 
+            if not disk.available:
+                logger.debug(
+                    f'Ignoring disk {disk.path}. Disk is unavailable due to {disk.rejected_reasons}'
+                )
+                continue
```

Comment:
@jschmid1 Sorry to bother you. Do you remember if we're going to create idempotency issues if we ignore unavailable disks when creating drive groups? I remember we had some discussions for the deepsea implementation back then, but I unfortunately no longer remember how we resolved and what caused them back then.

Reply:
Hey @sebastian-philipp, I'm not @jschmid1 ;-) but I _did_ do some work on this area of DeepSea's drive groups recently. There's some more detail in https://github.com/SUSE/DeepSea/pull/1877, but the summary is that shared DB devices are listed as unavailable, and if you ignore those devices, when _replacing_ an OSD whose DB is on a shared device, the DB of the _new_ OSD doesn't get put on the shared device and you end up with a standalone OSD.

Reply:
One further point - the DeepSea code _does_ ignore disks with GPT headers (it has its own separate check for that). Maybe it's worth doing something similar here, i.e. explicitly ignore only those disks, rather than ignoring all allegedly unavailable disks?

Reply:
@tserong The cephadm's OSD specs share the same code base. And I think they share the same issues indeed.

Hi @andrewschoen  I think Tim's DeepSea PR is a direct follow up on your #40201 . Seems that marking them as unavailable doesn't really solve this completely. Do you think we can actually *hide* disks with GPT headers from the inventory?